variables.tf

```diff
@@ -21,8 +21,14 @@ variable "allowed_aws_services_for_sns_published" {
 
 variable "kms_master_key_id" {
   type        = string
-  description = "The ID of an AWS-managed customer master key (CMK) for Amazon SNS or a custom CMK"
-  default     = "alias/aws/sns"
+  description = "The ID of an AWS-managed customer master key (CMK) for Amazon SNS or a custom CMK. If left empty, SNS Topic will have Encryption disabled."
+  default     = ""
+}
+
+variable "use_default_kms_master_key" {
+  type        = bool
+  description = "If set to true, then alias/aws/sns will be used in kms_master_key_id attribute of aws_sns_topic resource."
+  default     = false
 }
```

Comment:
This would break backwards compatibility and it seems like a less secure default. If this is introduced, I'd feel more comfortable if it defaulted to `true`.

Reply:
@azec-pdx I don't understand the difference between these two approachers:

```
kms_master_key_id = var.use_default_kms_master_key == true ? "alias/aws/sns" : var.kms_master_key_id
```

and just 
```
kms_master_key_id = var.kms_master_key_id

variable "kms_master_key_id" {
  type        = string
  default     = "alias/aws/sns"
}
```

If you want to use the default, don't change the variable (which has "alias/aws/sns" as the default).
If you want to provide your own, set it when you instantiate the module

```
kms_master_key_id = "<my_key_id>"
```

to me it looks the same as 
```
kms_master_key_id = var.use_default_kms_master_key == true ? "alias/aws/sns" : var.kms_master_key_id
```

without adding `var.use_default_kms_master_key`

Reply:
note that you could set `kms_master_key_id` to any value: your own key ID, an empty string or `null` to disable encryption

Reply:
To @nitrocode 's comment:
> This would break backwards compatibility and it seems like a less secure default. If this is introduced, I'd feel more comfortable if it defaulted to `true`.

Did you read comments of the people in the previous PR [#32](https://github.com/cloudposse/terraform-aws-sns-topic/pull/32) (linked at the top) that was closed? While it is not backward compatible, I am quite positive that there is a larger number of use-cases within AWS where security concerns are addressed via IAM permissions to SNS topic operations - especially within the boundaries of single AWS account and other AWS services interacting with SNS. 

With default key `alias/aws/sns` being imposed default, the assumption is that everyone will configure IAM service roles etc. to be able to work with that default KMS key. People generally don't want to do this, unless they already have a heavy use-case for using encryption on SNS and already provide customer's KMS key. What I am saying is that the effort on IAM side imposed by this default is worth only if you already know you need encryption and already are invested to do it right with your own KMS key. 

But that said, I am not opposed to this being `true` for keeping that backward compatibility for those using module (that is fair consideration).

Reply:
To @aknysh 'c comment:
> I don't understand the difference between these two approachers: ...

The way things are right now - you have to pass in `kms_master_key_id = ""` if you want to disable encryption.
I think that this is bad practice - having to be explicit with empty string variable to achieve opt-in/out of the feature. 
What I wanted to achieve is to be lazy if you don't want encryption (with both defaults), little explicit if you want at least some encryption (by setting `use_default_kms_master_key = true` ), more explicit if you want encryption and have your own KMS key instead ( `kms_master_key_id= "<KMS_ARN>"` )

Maybe the compromise would be to have a feature flag `use_encryption` and then before it is passed to resource, combine it with `kms_master_key_id` value: 

`use_encryption = true && (kms_master_key_id == "" | null) ==> use default key "alias/aws/sns"` as fallback
`use_encryption = true && (kms_master_key_id == <HAS_VALUE>) ==> try to use customer's key`
`use_encryption = false ==> use "" in resource for kms_master_key_id ==> no encryption`

Reply:
use a local for the evaluation of custom kms ley or not and leave the variable as it was with the default. 
you can add a new variable if you want too

Reply:
Good idea, we do that a lot in some internal stuff. I don't know why I didn't remember to apply same idea here.